Banff changes (#1012)
ava-labs/avalanche-docs: docs/specs/banff-changes.md

# Banff Changes

> This document specifies changes in Avalanche “Banff”, which was pre-released in AvalancheGo v1.8.x and will be finalized in AvalancheGo v1.9.x, from Avalanche Apricot Phase 5 (AvalancheGo v1.7.x).

Reviewer: This line doesn't really make a ton of sense to me.
```suggestion
This document specifies the changes in Avalanche “Banff”, which will be released in AvalancheGo v1.9.x.
```

## Block Changes

### Apricot

Apricot allows the following block types with the following content:

- _Standard Blocks_ may contain multiple transactions of the following types:
  - CreateChainTx
  - CreateSubnetTx
  - ImportTx
  - ExportTx
- _Proposal Blocks_ may contain a single transaction of the following types:
  - AddValidatorTx
  - AddDelegatorTx
  - AddSubnetValidatorTx
  - RewardValidatorTx
  - AdvanceTimeTx
- _Options Blocks_, i.e. _Commit Block_ and _Abort Block_ do not contain any transactions.

Each block has a header containing:

- ParentID
- Height

> ### Blueberry
> 
> Blueberry allows the following block types with the following content:

Reviewer: ```suggestion
### Banff

Banff allows the following block types with the following content:
```

	// Metadata, inputs and outputs
	BaseTx `serialize:"true"`
	// Describes the validator
	Validator validator.Validator `serialize:"true" json:"validator"`
	// ID of the subnet this validator is validating
	Subnet ids.ID `serialize:"true" json:"subnet"`
	// Where to send staked tokens when done validating
	Stake []*avax.TransferableOutput `serialize:"true" json:"stake"`
	// Where to send staking rewards when done validating
	RewardsOwner fx.Owner `serialize:"true" json:"rewardsOwner"`
}
```

#### New TypeIDs

```go
ApricotProposalBlock = 0
ApricotAbortBlock = 1
ApricotCommitBlock = 2
ApricotStandardBlock = 3
ApricotAtomicBlock = 4

secp256k1fx.TransferInput = 5
secp256k1fx.MintOutput = 6
secp256k1fx.TransferOutput = 7
secp256k1fx.MintOperation = 8
secp256k1fx.Credential = 9
secp256k1fx.Input = 10
secp256k1fx.OutputOwners = 11

AddValidatorTx = 12
AddSubnetValidatorTx = 13
AddDelegatorTx = 14
CreateChainTx = 15
CreateSubnetTx = 16
ImportTx = 17
ExportTx = 18
AdvanceTimeTx = 19
RewardValidatorTx = 20

stakeable.LockIn = 21
stakeable.LockOut = 22

RemoveSubnetValidatorTx = 23
TransformSubnetTx = 24
AddPermissionlessValidatorTx = 25
AddPermissionlessDelegatorTx = 26

> BlueberryProposalBlock = 27
> BlueberryAbortBlock = 28
> BlueberryCommitBlock = 29
> BlueberryStandardBlock = 30

Reviewer: ```suggestion
EmptyProofOfPossession = 27
BLSProofOfPossession   = 28

BanffProposalBlock = 29
BanffAbortBlock = 30
BanffCommitBlock = 31
BanffStandardBlock = 32
```
